control/modelsimp.py

```diff
@@ -368,38 +369,132 @@ def minreal(sys, tol=None, verbose=True):
     return sysr
 
 
-def era(YY, m, n, nin, nout, r):
-    """Calculate an ERA model of order `r` based on the impulse-response data
-    `YY`.
+def era(arg, r, m=None, n=None, dt=True, transpose=False):
+    r"""era(YY, r)
 
-    .. note:: This function is not implemented yet.
+    Calculate an ERA model of order `r` based on the impulse-response data.
+
+    This function computes a discrete time system
+
+    .. math::
+
+        x[k+1] &= A x[k] + B u[k] \\\\
+        y[k] &= C x[k] + D u[k]
+
+    for a given impulse-response data (see [1]_).
+
+    The function can be called with 2 arguments:
+
+    * ``sysd, S = era(data, r)``
+    * ``sysd, S = era(YY, r)``
+
+    where `response` is an `TimeResponseData` object, and `YY` is 1D or 3D
+    array and r is an integer.
 
     Parameters
     ----------
-    YY: array
-        `nout` x `nin` dimensional impulse-response data
-    m: integer
-        Number of rows in Hankel matrix
-    n: integer
-        Number of columns in Hankel matrix
-    nin: integer
-        Number of input variables
-    nout: integer
-        Number of output variables
-    r: integer
-        Order of model
+    YY : array_like
+        impulse-response data from which the StateSpace model is estimated,
+        1D or 3D array.
```

Comment:
impulse → Impulse (start sentence with a capital letter) and try to fit on one line (if less than 75 characters).

Reply:
Done.